File: java/acronym/src/main/java/Acronym.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Acronym {
    public static String generate(String phrase) {
        List<String> words = Arrays.asList(phrase.split("(\\W+|[a-z](?=[A-Z]))"));

        return words.stream()
                .map(word -> word.substring(0, 1))
                .collect(Collectors.joining())
                .toUpperCase();
    }
}

File: ecmascript/acronym/acronym.js
export default {
  parse: phrase => {
    const words = phrase.split(/\s+|-|[a-z](?=[A-Z])/g);
    return words
            .map(word => word.substring(0, 1))
            .reduce((acronym, next) => acronym + next)
            .toUpperCase();
  },
};
